README.md

```diff
@@ -32,6 +33,73 @@ pip install scikeras
 
 The only dependencies are `scikit-learn>=0.22` and `TensorFlow>=2.2.0`.
 
+## Transitioning from `tf.keras.wrappers.scikit_learn`
+
+SciKeras is largely backwards compatible with the existing wrappers. For most cases, you can just change your import statement from:
+
+```python
+from tensorflow.keras.wrappers.scikit_learn import KerasClassifier, KerasRegressor  # from
+from scikeras.wrappers import KerasClassifier, KerasRegressor  # to
+```
+
+SciKeras does however have some backward incompatible changes:
+
+### Automatic one-hot encoding of targets for categorical crossentropy losses
+
+SciKeras will not longer implicitly inspect your Model's loss function to determine if
+it needs to one-hot encode your target to match the loss function. Instead, you must explicitly
+pass your loss function to the constructor:
+
+```python
+clf = KerasClassifier(loss="categorical_crossentropy")
+```
+
+### Removal of `**kwargs` from fit and predict
+
+In a future release of SciKeras, `**kwargs` will be removed from fit and predict. To future
+proof your code, you should instead declare these parameters in your constructor:
+
+```python
+clf = KerasClassifier(batch_size=32)
+```
+
+Or to declare separate values for `fit` and `predict`:
+
+```python
+clf = KerasClassifier(fit__batch_size=32, predict__batch_size=32)
+```
+
+### Renaming of `build_fn` to `model`
+
+SciKeras renamed the constructor argument `build_fn` to `model`. In a future release,
+passing `build_fn` as a _keyword_ argument will raise a `TypeError`. Passing it as a positional
+argument remains unchanged. You can make the following change to future proof your code:
+
+```python
+clf = KerasClassifier(build_fn=...)  # from
+clf = KerasClassifier(model=...)  # to
+```
+
+### Default arguments in `build_fn`/`model`
+
+SciKeras will no longer introspect your callable `model` for _user defined_ parameters. You
+must now "declare" them as keyword arguments to the constructor if you want them to be
+tunable parameters (i.e. settable via `set_params`):
```

Comment:
Did Keras ever introspect?

On master, doesn't parameter routing remove the need for declaration?

Reply:
> Did Keras ever introspect?

Yes, in Keras the following works:

```python3
def get_model(my_param=123):
    ...
clf = KerasClassifier(get_model)
clf.set_params(my_param=321)
```

In SciKeras, this would not work. You would have to use `KerasClassifier(get_model, my_param=...)`.
But if `set_params` is not being called, no changes are needed.

> On master, doesn't parameter routing remove the need for declaration?

I didn't understand the question, could you rephrase? Sorry...

Reply:
> I didn't understand the question, could you rephrase? Sorry...

I thought SciKeras supported prefixing `model__` to various parameters to route to the model building function. That is, I though this worked:

``` python
def get_model(foo=123):
    ...
clf = KerasClassifier(get_model)
clf.set_params(model__foo=456)
```

I thought SciKeras had deprecated setting parameters without a `model__` prefix (e.g., `clf.set_params(foo=456)`)... or did we walk that back?

Reply:
It does, but it's not _required_ unless there is a collision (e.g. your model building function for whatever reason has a parameter called `batch_size`). And since the previous implementation does not support that, I don't see why I should be mentioned in the transition guide.